src/main/java/org/apache/commons/text/CaseUtils.java

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * <p>Case manipulation operations on Strings that contain words.</p>
+ *
+ * <p>This class tries to handle <code>null</code> input gracefully.
+ * An exception will not be thrown for a <code>null</code> input.
+ * Each method documents its behaviour in more detail.</p>
+ *
+ * @since 1.0
+ */
+public class CaseUtils {
+
+    /**
+     * <p><code>CaseUtils</code> instances should NOT be constructed in
+     * standard programming. Instead, the class should be used as
+     * <code>CaseUtils.toCamelCase("foo bar", true, new char[]{'-'});</code>.</p>
+     * <p>
+     * <p>This constructor is public to permit tools that require a JavaBean
+     * instance to operate.</p>
+     */
+    public CaseUtils() {
+        super();
+    }
+
+    // Camel Case
+    //-----------------------------------------------------------------------
+    /**
+     * <p>Converts all the delimiter separated words in a String into camelCase,
+     * that is each word is made up of a titlecase character and then a series of
+     * lowercase characters. The  </p>
+     *
+     * <p>The delimiters represent a set of characters understood to separate words.
+     * The first non-delimiter character after a delimiter will be capitalized. The first String
+     * character may or may not be capitalized and it's determined by the user input for capitalizeFirstLetter
+     * variable.</p>
+     *
+     * <p>A <code>null</code> input String returns <code>null</code>.
+     * Capitalization uses the Unicode title case, normally equivalent to
+     * upper case.</p>
+     *
+     * <pre>
+     * CaseUtils.toCamelCase(null, false)                                 = null
+     * CaseUtils.toCamelCase("", false, *)                                = ""
+     * CaseUtils.toCamelCase(*, false, null)                              = *
+     * CaseUtils.toCamelCase(*, true, new char[0])                        = *
+     * CaseUtils.toCamelCase("To.Camel.Case", false, new char[]{'.'})     = "toCamelCase"
+     * CaseUtils.toCamelCase(" to @ Camel case", true, new char[]{'@'})   = "toCamelCase"
+     * CaseUtils.toCamelCase(" @to @ Camel case", false, new char[]{'@'}) = "toCamelCase"
+     * </pre>
+     *
+     * @param str  the String to be converted to camelCase, may be null
+     * @param capitalizeFirstLetter boolean that determines if the first character of first word should be title case.
+     * @param delimiters  set of characters to determine capitalization, null and/or empty array means whitespace
+     * @return camelCase of String, <code>null</code> if null String input
+     */
+    public static String toCamelCase(String str, boolean capitalizeFirstLetter, final char... delimiters) {
```

Comment:
Yes the boolean should be final not sure why I missed that . The input string is not final as you mentioned  I change it to lower case inside the method .